### PR TITLE
[#165] [#192] Add tests for specific cases of local references

### LINKED ISSUE #165: Test all variants of local references
# Clarification and motivation

Let's extend the bats tests a bit to test some basic local references:
* "self links" to an anchor within the same file: `#anchor`
* "relative links" with and without anchors: `dir/file.md`, `dir/file.md#anchor`, `./dir/file.md`, `dir/./file.md`, `../dir/file.md`
* "absolute links" with and without anchors: `/dir/file.md`, `dir/file.md#anchor`, `/dir/./file.md`
* Test all of this when the root directory matches the current directory (i.e. the default `--root .`) **and** when it doesn't (i.e. `--root somedir`)

The tests in `tests/golden/check-anchors` already cover the specific case of relative links with anchors when `--root .`, but I think those tests should be replaced with a more complete suite for _all_ kinds of local references

# Acceptance criteria

* We have bats tests for all variants of local references

### LINKED ISSUE #192: Test case-sensitivity
# Clarification and motivation

One more quirk I just came across while reading the docs of [`equalFilePath`](https://hackage.haskell.org/package/filepath-1.4.100.0/docs/System-OsPath-Posix-Internal.html#v:equalFilePath): 

On linux, filepaths are case-sensitive. So, if you have a file named `a` and a link to `A`, the link is invalid.
Both GitHub and GitLab's URIs are also case-sensitive:
* https://github.com/serokell/xrefcheck/blob/master/README.md <- this link works
* https://github.com/serokell/xrefcheck/blob/master/readme.md <- this doesn't

So we should have tests for case-sensitivity.

Windows paths, however, are case-**insensitive**. So these tests will also be useful for #164

# Acceptance criteria

* We have tests for case-sensitivity.

### PR DESCRIPTION
**Pull request by Sorokin-Anton**
## Description
Problem: there is no tests for links to directories and non-markdown files. Also we aren't testing that links are case sensetive

Solution: add such tests to `check-local-refs`, also check that only links to directory are allowed to have trailing slash,
e.g. `a.md/` is bad and `dir1/` is ok

## Related issue(s)

Fixes #165 
Fixes #192 

## :white_check_mark: Checklist for your Pull Request

Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](https://github.com/serokell/xrefcheck/tree/master/README.md)
    - Haddock

- Public contracts
  - [x] Any modifications of public contracts comply with the [Evolution
  of Public Contracts](https://www.notion.so/serokell/Evolution-of-Public-Contracts-2a3bf7971abe4806a24f63c84e7076c5) policy.
  - [x] I added an entry to the [changelog](https://github.com/serokell/xrefcheck/tree/master/CHANGES.md) if my changes are visible to the users
        and
  - [x] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](https://github.com/serokell/style/blob/master/haskell.md).

### CODE REVIEW

File: tests/golden/check-local-refs/dir1/dir2/d2f1.md [hunk 1]
```diff
@@ -9,23 +9,32 @@
...
+[existing-dir-rel-1](..)
+[existing-dir-rel-2](../dir2)
+[existing-dir-rel-3](../dir2/)
 
 [existing-anchor-rel-1](d2f2.md#existing-anchor-d2f2)
 [existing-anchor-rel-2](./d2f2.md#existing-anchor-d2f2)
 [existing-anchor-rel-3](../dir2/../d1f1.md#existing-anchor-d1f1)
 
 [bad-file-rel](../a/b/c/unexisting-file.md)
+[bad-slash-file-rel](d2f2.md/)
```
Comment (dcastro) [inline tests/golden/check-local-refs/dir1/dir2/d2f1.md:26]: This link should be valid, both github and gitlab correctly link to `[a](./README.md/)`. I'll create an issue for this.

File: tests/golden/check-local-refs/dir1/dir2/d2f1.md [hunk 2]
```diff
@@ -9,23 +9,32 @@
...
+[bad-casing-file-rel](D2F2.md/)
+[bad-casing-folder-rel](../DIR2)
 
 [bad-anchor-rel-1](d2f2.md#bad-anchor)
 [bad-anchor-rel-2](unexisting-file.md#bad-anchor)
 
 # Absolute links
 Should be correct when root is `/tests/golden/check-local-refs`:
 [file-abs-1](/dir1/./d1f1.md)
```
Comment (dcastro) [inline tests/golden/check-local-refs/dir1/dir2/d2f1.md:35]: Let's also add an absolute link + to a file + with a trailing slash

`[file-abs-2](/dir1/./d1f1.md/)`

File: tests/golden/check-local-refs/dir1/dir2/d2f1.md (hunk 1 above)
Comment (Sorokin-Anton) [reply tests/golden/check-local-refs/dir1/dir2/d2f1.md:26]: Ok, I left this untouched

File: tests/golden/check-local-refs/dir1/dir2/d2f1.md (hunk 2 above)
Comment (Sorokin-Anton) [reply tests/golden/check-local-refs/dir1/dir2/d2f1.md:39]: Done

**Pull request merged**